myapp/app/controllers/application_controller.rb

```diff
@@ -1,2 +1,38 @@
 class ApplicationController < ActionController::Base
+
+  around_action :switch_locale
+
+  def switch_locale(&action)
+    locale = params[:locale] || I18n.default_locale
+    I18n.with_locale(locale, &action)
+  end
+
+  # keep locale in urls
+  def default_url_options
+    { locale: I18n.locale }
+  end
+
+
+  # unless Rails.env.development?
+    rescue_from Exception,                        with: :_render_500
+    rescue_from ActiveRecord::RecordNotFound,     with: :_render_404
+    rescue_from ActionController::RoutingError,   with: :_render_404
+  # end
+
+  def routing_error
+    raise ActionController::RoutingError, params[:path]
+  end
+
+  private
+    def _render_404(e = nil)
```

Comment:
Usually, in ruby, it is recommended that variable names be at least 3 characters. 
```suggestion
    def _render_404(exception = nil)
```

Reply:
fixed